projects/superfluid.js

```diff
@@ -63,20 +60,44 @@ async function getChainBalances(allTokens, chain, block, isVesting, api) {
       symbol,
       isNativeAssetSuperToken,
     } = allTokens[i];
-    let underlyingTokenBalance =
-      (totalSupply * 10 ** (underlyingToken || { decimals: 18 }).decimals) /
-      10 ** decimals;
+
     // Accumulate to balances, the balance for tokens on mainnet or sidechain
     let prefixedUnderlyingAddress = underlyingAddress;
-    // if (!underlyingToken && underlyingTokenBalance/1e24 > 1) sdk.log(name, symbol, chain, Math.floor(underlyingTokenBalance/1e24))
-    // if (isNativeAssetSuperToken) prefixedUnderlyingAddress = chain + ':' + underlyingAddress
     if (
-      !underlyingToken ||
+      underlyingAddress &&
       blacklistedSuperTokens.has(underlyingAddress.toLowerCase())
     )
-      return;
-    api.add(prefixedUnderlyingAddress, underlyingTokenBalance);
-  });
+      continue;
+
+    // ALEPH custom logic (Base and Avalanche): no underlying; circulating = totalSupply - locker balance
+    if (
+      (chain === 'base' || chain === 'avax') &&
+      id.toLowerCase() === ALEPH_SUPERTOKEN
+    ) {
+      const lockerHolding = await api.call({ abi: 'erc20:balanceOf', target: id, params: [ALEPH_LOCKER] });
+      const circulating = Math.max(0, totalSupply - lockerHolding);
+      api.add(id, circulating);
+      continue;
+    }
```

Comment:
@waynebruce0x Double checked and there was indeed an issue with circulation of the ALEPH token , it has custom logic that basicly has a holder of the whole total supply on side chains for the bridged tokens from mainnet , so the actual balance of ALEPH tokens that are Super tokens and available for users is totalSupply - locker balance , fixed it here , should be good now